### DISCUSSION TITLE
R stats - t.test() vs glm()

### DISCUSSION THREAD

**Discussion by kubu4** [RobertsLab/resources#1498] [Q&A]
@sr320 and I were playing with some analysis today and compared these two functions. Similar, but different, results. Conclusions about the data wouldn't have been affected, but we're curious what the difference is between the two and why you use one instead of the other.

Anyone have any thoughts/opinions on this?

**Comment by kubu4** [comment]
Just bumping this to see if anyone had any feedback on this.

**Reply by sr320** [reply]
And for more context what is proper use for qPCR where there is one treatment.

**Comment by aspencoyle** [comment]
Just saw this! Okay, so t.test() does a t-test, obviously, but glm() is essentially running a linear regression (glm = generalized linear model). If you want to incorporate random effects, you can use glmer().

My understanding, which could be way off, is that when you've got one treatment with levels (say, looking at qPCR results with control and elevated-temp. mussels), you want to use a t-test. If you've got a continuous variable (say, taking samples from intertidal mussels throughout the summer and comparing ambient water temperature to qPCR results), a linear model would be better. Linear models are also more flexible - you can add in several predictors, add random effects, and so on!

**Reply by kubu4** [reply]
Thanks. One of the reasons this came up is that FISH541 used `glm()` in their qPCR analysis, but @sr320 and I were using `t.test()` instead, as this is what we were more familiar with. Any idea why FISH541 would've opted for `glm()`, as the comparison was the same - control vs. heat-stressed.

**Reply by aspencoyle** [reply]
I'm not sure, honestly! Could definitely have just been a mix-up, that analysis was ran at the end of the quarter and we were rushing through things a bit
